workers/unity/Packages/com.improbable.gdk.core/Worker/WorkerBase.cs

```diff
@@ -1,108 +1,158 @@
 using System;
+using System.Collections.Generic;
+using System.Linq;
+using Improbable.Gdk.Core.Components;
 using Improbable.Worker;
 using Unity.Entities;
 using UnityEngine;
+using Entity = Unity.Entities.Entity;
 
 namespace Improbable.Gdk.Core
 {
     public abstract class WorkerBase : IDisposable
     {
-        public World World { get; private set; }
-
-        public MutableView View { get; private set; }
-
         public Connection Connection { get; private set; }
 
-        public Vector3 Origin { get; private set; }
+        public Vector3 Origin { get; }
 
-        public string WorkerId { get; private set; }
+        public string WorkerId { get; }
 
         public abstract string GetWorkerType { get; }
 
-        public EntityGameObjectLinker EntityGameObjectLinker { get; private set; }
+        
 
         public bool UseDynamicId { get; protected set; }
 
-        protected WorkerBase(string workerId, Vector3 origin) : this(workerId, origin, new LoggingDispatcher())
-        {
-        }
-
-        protected WorkerBase(string workerId, Vector3 origin, ILogDispatcher loggingDispatcher)
-        {
-            if (string.IsNullOrEmpty(workerId))
-            {
-                WorkerId = GenerateDynamicWorkerId();
-
-                UseDynamicId = true;
-            }
-            else
-            {
-                WorkerId = workerId;
-            }
+        public readonly EntityManager EntityManager;
+        public const long WorkerEntityId = -1337;
+        public readonly Entity WorkerEntity;
 
-            World = new World(WorkerId);
-            WorkerRegistry.SetWorkerForWorld(this);
+        public readonly ILogDispatcher LogDispatcher;
 
-            View = new MutableView(World, loggingDispatcher);
-            Origin = origin;
+        public readonly Type[] Systems = new[]
+        {
+            typeof(SpatialOSReceiveSystem),
+            typeof(SpatialOSSendSystem),
+            typeof(CleanReactiveComponentsSystem)
+        };
 
-            EntityGameObjectLinker = new EntityGameObjectLinker(World, View);
-        }
+        private readonly Dictionary<long, Entity> entityMapping;
 
-        public void Dispose()
+        protected WorkerBase(string workerId, ConnectionConfig config, Vector3 origin, EntityManager entityManager)
         {
-            WorkerRegistry.UnsetWorkerForWorld(this);
-            View.Dispose();
-            World.Dispose();
-        }
+            FindTranslationUnits();
+            // TODO addAllCommandRequestSenders(WorkerEntity, WorkerEntityId);
+            entityMapping = new Dictionary<long, Entity>();
 
-        public virtual void Connect(ConnectionConfig config)
-        {
+            LogDispatcher = new ForwardingDispatcher();
```

Comment:
This feels like a strong opinion in the core - this should be left up to the user imo